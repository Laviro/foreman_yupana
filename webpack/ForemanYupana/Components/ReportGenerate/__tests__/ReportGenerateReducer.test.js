import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import {
  REPORT_GENERATE_LOGS_POLLING,
  REPORT_GENERATE_PROCESS_START,
  REPORT_GENERATE_PROCESS_STOP,
  REPORT_GENERATE_PROCESS_FINISH,
} from '../ReportGenerateConstants';
import reducer from '../ReportGenerateReducer';
import {
  logs,
  completed,
  processID,
  pollingProcessID,
  status,
} from '../ReportGenerate.fixtures';

const fixtures = {
  'should return the initial state': {},
  'should handle REPORT_GENERATE_LOGS_POLLING_START': {
    action: {
      type: REPORT_GENERATE_LOGS_POLLING,
      payload: {
        pollingProcessID,
      },
    },
    'should handle REPORT_GENERATE_LOGS_POLLING': {
      action: {
        type: REPORT_GENERATE_LOGS_POLLING,
        payload: {
          logs,
          completed,
        },
      },
      'should handle REPORT_GENERATE_PROCESS_START': {
        action: {
          type: REPORT_GENERATE_PROCESS_START,
          payload: {
            processID,
            status,
          },
        },
      },
      'should handle REPORT_GENERATE_PROCESS_STOP': {
        action: {
          type: REPORT_GENERATE_PROCESS_STOP,
          payload: {
            logs,
            completed,
            status,
          },
        },
      },
      'should handle REPORT_GENERATE_PROCESS_FINISH': {
        action: {
          type: REPORT_GENERATE_PROCESS_FINISH,
          payload: {
            status,
          },
        },
      },
    },
  },
};

describe('ReportGenerate reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
