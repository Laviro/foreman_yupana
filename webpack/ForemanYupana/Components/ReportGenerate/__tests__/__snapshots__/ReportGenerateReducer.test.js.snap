// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportGenerate reducer should handle REPORT_GENERATE_LOGS_POLLING_START 1`] = `
Object {
  "completed": undefined,
  "logs": undefined,
  "pollingProcessID": 0,
  "processID": 0,
  "status": "stopped",
}
`;

exports[`ReportGenerate reducer should return the initial state 1`] = `
Object {
  "completed": 0,
  "logs": Array [
    "No running process",
  ],
  "pollingProcessID": 0,
  "processID": 0,
  "status": "stopped",
}
`;
